chia/full_node/block_store.py

```diff
@@ -329,24 +359,23 @@ async def get_generator(self, header_hash: bytes32) -> Optional[SerializedProgra
 
         formatted_str = "SELECT block, height from full_blocks WHERE header_hash=?"
         async with self.db_wrapper.reader_no_transaction() as conn:
-            async with conn.execute(formatted_str, (self.maybe_to_hex(header_hash),)) as cursor:
-                row = await cursor.fetchone()
-                if row is None:
-                    return None
-                if self.db_wrapper.db_version == 2:
-                    block_bytes = zstd.decompress(row[0])
-                else:
-                    block_bytes = row[0]
-
-                try:
-                    return generator_from_block(block_bytes)
-                except Exception as e:
-                    log.error(f"cheap parser failed for block at height {row[1]}: {e}")
-                    # this is defensive, on the off-chance that
-                    # generator_from_block() fails, fall back to the reliable
-                    # definition of parsing a block
-                    b = FullBlock.from_bytes(block_bytes)
-                    return b.transactions_generator
+            row = await execute_fetchone(conn, formatted_str, (self.maybe_to_hex(header_hash),))
```

Comment:
this is a somewhat unrelated change to make `get_generator()` just have a single round-trip to the DB thread (by using `execute_fetchone()`)